src/App.tsx

```diff
@@ -1,61 +1,122 @@
+import { useState } from 'react';
 import './App.scss';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+const initialTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [selectedUser, setSelectedUser]
+  = useState< number | undefined>(undefined);
+  const [inputValue, setInputValue] = useState('');
+  const [todos, setTodos] = useState(initialTodos);
+  const [userError, setUserError] = useState('');
+  const [inputError, setInputError] = useState('');
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+
+    if (!selectedUser) {
+      setUserError('Please choose a user');
+
+      return;
+    }
+
+    if (!inputValue.trim()) {
+      setInputError('Please enter a title');
+
+      return;
+    }
```

Comment:
The user should see both errors if the user is not selected and the title is empty at the same time. 
Now it shows only one error at a time. Try to refactor it a bit.